server.js

```diff
@@ -1,33 +1,142 @@
+*/
+
+app.get('/books', (request, response) => {
+  const { author, title, language, averagerating, page } = request.query;
+
+  if (author) {
+    //tolocaleLowerCase() sets the data in author array data to lowercase and also the author that is being queried so they will match
+    const authorResults = booksData.filter((item) => item.authors.toLocaleLowerCase().includes(author.toLocaleLowerCase()));
+    if(authorResults.length === 0) {
+      response.status(404).json("Sorry that author doesn't exist.");
+    }
+      response.json(authorResults);
+  }
+  
+  if (title) {
+    const titleResults = booksData.filter((item) => item.title.toString().toLocaleLowerCase().includes(title.toLocaleLowerCase()));
+    if(titleResults.length === 0) {
+      response.status(404).json("Sorry that book title doesn't exist.");
+    } 
+    response.json(titleResults);
+  }
+
+  if (language) {
+    const languageResults = booksData.filter((item) => item.language_code === language);
+    if(languageResults.length === 0) {
+      response.status(404).json("Sorry we don't have any books in that language.");
+    } 
+    response.json(languageResults);
+  }
+
+  // "+" on averagerating turns the string in to a number.
+  if (averagerating) {
+    const aveRatingResults = booksData.filter((item) => item.average_rating === +averagerating);
+    if(aveRatingResults.length === 0) {
+      response.status(404).json("Sorry, that rating doesn't exist.");
+    }
+     response.json(aveRatingResults);    
+  } 
+
+  if(page) {
+    let pageSize=50;
+    // Determining the index number of the first book on the page, as page is storing a number between 1-10 that the user requests. This number is minused by 1 and times by 50 to give the starting index number to be used in the slice.
+    let startIndex = (page-1)*pageSize;
+    // Then we use that startIndex as the starting point of the slice and  then end startIndex plus 50 to slice the last index from the array of books
+    let slice = booksData.slice(startIndex, startIndex+pageSize);
+    // Showing the result of the slice
+    response.json(slice);
+  }
+
+  response.json(booksData);
+});
+
+/* ---- Second endpoint ----
+Getting a specific book based on the id added to the endpoint. 
+:id is a variable that will be the value the user has specified on the endpoint e.g. 3. Use params to get the id/value the user has specified in endpoint and put that into the bookId variable. We then use that when finding the id in the array by comparing the bookID number in the array with the id the user has specified on the endpoint. 
+If statement for if the length of the specificBook variable/array is 0 (i.e. the author doesn't exist) then return the error message, else translate the bookID element from the array into json so the data it's readable on the endpoint.*/
+//http://localhost:8081/books/10
+app.get('/books/:id', (request, response) => {
+  const bookID  = request.params.id;
```

Comment:
To be consistent, you can destructure this as well:
`const { bookID }  = request.params;`